tesla-hero-slider/src/components/slider.js

```diff
@@ -0,0 +1,104 @@
+import React, {Component} from 'react';
+import Slide from './slide';
+import SliderNavigation from './slider-navigation';
+import slides from './sliders-data';
+import mouseImg from './../assets/images/mouse.svg';
+import './../styles/scss/components/slider.css';
+
+const ANIMATION_PHASES = {
+	PENDING: 'PENDING',
+	STOP: 'STOP'
+};
+
+class Slider extends Component {
+
+	state = {
+		activeSlide: 0,
+		animationForward: true,
+		slidesCount: slides.length,
+		animationState: null
+	};
+
+	slider = {
+		header: '',
+		content: ''
+	};
+
+	componentDidMount() {
+		this.setState({
+			activeSlide: 3
+		});
+		this.setAnimationState(ANIMATION_PHASES.PENDING);
+
+		this.slider.header = document.querySelector('.tesla-header');
+		this.slider.content = document.querySelector('.tesla-slider')
+
+		document.body.addEventListener('wheel', this.handleScroll.bind(this));
+	}
+
+	componentWillUnmount() {
+		document.body.removeEventListener('wheel', this.handleScroll);
+	}
+
+	setAnimationState = animationState => this.setState({animationState});
+
+	setActiveSlide(slideId) {
+		this.setState({
+			activeSlide: slideId,
+			animationForward: this.state.activeSlide < slideId ? true : false
```

Comment:
This doesn't handle the case when a user is going from the first to last and from the last to first. Maybe we don't need it? Do we need a loop?